virttest/utils_libvirt/libvirt_ceph_utils.py

```diff
@@ -0,0 +1,153 @@
+                                   debug=True)
+            disk_auth_dict = {"auth_user": ceph_auth_user,
+                              "secret_type": auth_sec_usage_type,
+                              "secret_uuid": auth_sec_uuid}
+            device_source = "rbd:%s:mon_host=%s:keyring=%s" % (ceph_disk_name,
+                                                               ceph_mon_ip,
+                                                               key_file)
+        else:
+            device_source = "rbd:%s:mon_host=%s" % (ceph_disk_name, ceph_mon_ip)
+        logging.debug("device source is: %s", device_source)
+        # clean up image file if exists
+        ceph.rbd_image_rm(ceph_mon_ip, ceph_disk_name.split('/')[0],
+                          ceph_disk_name.split('/')[1], keyfile=rbd_key_file)
+
+        #Create necessary image file if not exists
+        if img_file is None:
+            img_file = os.path.join(data_dir.get_data_dir(),
+                                    "%s_test.img" % vm.name)
+            # Create an local image and make FS on it.
+            disk_cmd = ("qemu-img create -f %s %s %s" %
+                        (device_format, img_file, storage_size))
+            process.run(disk_cmd, ignore_status=False, shell=True)
+        # Convert the image to remote ceph storage
+        disk_path = ("rbd:%s:mon_host=%s" %
+                     (ceph_disk_name, ceph_mon_ip))
+        if ceph_client_name and ceph_client_key:
+            disk_path += (":id=%s:key=%s" %
+                          (ceph_auth_user, ceph_auth_key))
+        rbd_cmd = ("rbd -m %s %s info %s 2> /dev/null|| qemu-img convert -O"
+                   " %s %s %s" % (ceph_mon_ip, key_opt, ceph_disk_name,
+                                  device_format, img_file, disk_path))
+        process.run(rbd_cmd, ignore_status=False, shell=True, verbose=True)
+
+        # Disk related config
+        disk_src_dict = {"attrs": {"protocol": "rbd",
+                                   "name": ceph_disk_name},
+                         "hosts":  [{"name": ceph_mon_ip,
+                                     "port": ceph_host_port}]}
+        # Create disk xml with given config
+        vmxml = vm_xml.VMXML.new_from_dumpxml(vm.name)
+        disk_xml = Disk(type_name='network')
+        disk_xml.device = device
+        disk_xml.target = {"dev": device_target, "bus": device_bus}
+        driver_dict = {"name": "qemu", "type": device_format}
+        disk_xml.driver = driver_dict
+        disk_source = disk_xml.new_disk_source(**disk_src_dict)
+        if disk_auth_dict:
+            logging.debug("disk auth dict is: %s" % disk_auth_dict)
+            disk_xml.auth = disk_xml.new_auth(**disk_auth_dict)
+        disk_xml.source = disk_source
+        logging.debug("new disk xml is: %s", disk_xml)
+        if not keep_raw_image_as:
+            if hotplug:
+                attach_option = params.get("virt_device_attach_option", "--config")
```

Comment:
why we use '--config' here as the default "virt_device_attach_option"? we often call this kind of attach-device as 'cold plug' if '--config' added. Such as:
```
Bug 1216046 - guest which cold-plug too much memory device will disappear after restart libvirt
Bug 1559867 - When COLD PLUG multiple scsi hostdevs to an ACTIVE vm, their addresses will be duplicated
```

Reply:
virt_device_attach_option is need to be specified in cfg dependent on hot or cold plug.  If not "--config" will be set as default value.
But anyway, I can change to "--live"